src/runtime/pkg/govmm/qemu/qemu.go

```diff
@@ -2999,14 +2999,18 @@ func (config *Config) appendVGA() {
 	}
 }
 
-func (config *Config) appendKernel() {
+func (config *Config) appendKernel(logger QMPLog) {
 	if config.Kernel.Path != "" {
 		config.qemuParams = append(config.qemuParams, "-kernel")
 		config.qemuParams = append(config.qemuParams, config.Kernel.Path)
 
 		if config.Kernel.InitrdPath != "" {
 			config.qemuParams = append(config.qemuParams, "-initrd")
 			config.qemuParams = append(config.qemuParams, config.Kernel.InitrdPath)
+		} else {
+			if logger != nil {
+				logger.Infof("initrd path is empty, assuming IBM Z Secure Execution")
```

Comment:
This will be logged even on non IBM z architectures right? I think this needs to go in the architecture specific qemu implementaion file. Similar comment for the code below as well.

Reply:
I think this would be no such a case for other architectures due to `conf.HypervisorMachineType == QemuCCWVirtio` below. This conditional is already s390x specific.

Reply:
Hmm.. now I got your point. this will be logged for where kernel and image are set. I will change the code as you suggested. But others wouldn't be reached out because the conditional is located in where image and initrd are not set and a machine type is s390 specific. Thanks!

Reply:
Update: I've removed the log for `appendKernel()` because it should be architecture-agnostic and an error due to the misconfiguration of the parameters is unlikely to happen thanks to the validation code at a hypervisor level.